edb/lang/edgeql/compiler/setgen.py

```diff
@@ -765,3 +719,73 @@ def _check_cardinality(ctx):
     rptr.target = comp_ir_set
 
     return comp_ir_set
+
+
+def _get_computable_ctx(
+        *,
+        rptr: irast.Pointer,
+        source: irast.Set,
+        source_scls: s_nodes.Node,
+        inner_source_path_id: irast.PathId,
+        qlctx: context.ContextLevel,
+        ctx: context.ContextLevel) -> typing.ContextManager:
+    @contextlib.contextmanager
+    def newctx():
+        with ctx.new() as subctx:
+            subctx.class_view_overrides = {}
```

Comment:
Would it make sense to make `_get_computable_ctx` a method on the Context object?

Reply:
Not really.  I strive to keep `Context` as close to a dumb dataclass as possible with only very simple logic.